### ISSUE TITLE
vmagent: Add match_first_network support for docker service discovery

### ISSUE THREAD

**Issue by keslerm** [VictoriaMetrics/VictoriaMetrics#7398]
### Describe the bug

I would expect vmagent to honor the configuration value when passed

### To Reproduce

Create a prometheus config with `docker_sd_configs` and add the setting `match_first_network: true`

### Version

vmagent-20241021-132821-tags-v1.105.0-0-g05f6ea621d

### Additional information

Configuration for this can be found in the prometheus docs https://prometheus.io/docs/prometheus/latest/configuration/configuration/#docker_sd_config

**Comment by jiekun** [comment]
Thanks! It seems it's available since  https://github.com/prometheus/prometheus/pull/10490 and version 2.54.0 (2024-08-09). I'd like to label it as `enhancement` instead of `bug`.

It would be appreciated if anyone wishes to contribute to this feature.

**Comment by manthanguptaa** [comment]
Hey @jiekun, would love to make an attempt at this!

**Comment by jiekun** [comment]
@manthanguptaa Thanks for volunteering.

The service discovery codes are mostly placed under `lib/promscrape/discovery` path. In this issue you should look into the `lib/promscrape/discovery/docker/docker.go` file.

There's a `type SDConfig struct`, you can check and see if it align with our documentation [here](https://docs.victoriametrics.com/sd_configs/#docker_sd_configs). 

In this enhancement, we want to add `match_first_network` param into this struct. It should work just like how Prometheus does, which could be found [here](https://github.com/prometheus/prometheus/blob/51866b9feebe9a8cd71f6bf19681919f34d0ad5f/discovery/moby/docker.go#L254).

Please don't miss the [pull request](https://github.com/prometheus/prometheus/pull/10490) in Prometheus project, which contains some context and discussion as well.

Please let me know if you need more help on this.

**Comment by jiekun** [comment]
@manthanguptaa Hello! May I ask if you are working on this feature?

**Comment by jiekun** [comment]
Yeah sure no hurry. Just want to confirm you are taking this one. Please take your time!

**Comment by jiekun** [comment]
@keslerm Hi. I tried to implement this feature. Could you try with image tag:
```
victoriametrics/victoria-metrics:heads-feature-docker-sd-match-first-network-0-g42d597b22
victoriametrics/vmagent:heads-feature-docker-sd-match-first-network-0-g42d597b22
```
or build it from `feature/docker-sd-match-first-network` branch. See: https://docs.victoriametrics.com/#how-to-build-from-sources.

Please let me know if you need any help.

**Comment by f41gh7** [comment]
In order to mitigate changes introduced at the linked PR, service discovery configuration may require to set  `match_first_network: false`.  
 It may be needed  in the following cases:
1. if multiple `networks` attached to the containers and it's required to select one of networks depending on labels.
2. if multiple `networks` attached to the containers and first network by ascending sorting shouldn't be used.


 See the following example how to choose network based on relabeling:
```yaml
scrape_configs:
  - job_name: "docker-containers"
    docker_sd_configs:
      - host: unix:///var/run/docker.sock # You can also use http/https to connect to the Docker daemon.
       # change default value to false
        match_first_network: false
    relabel_configs:
      # Only keep containers that have a `prometheus-job` label.
      - source_labels: [__meta_docker_container_label_prometheus_job]
        regex: .+
        action: keep
      # Only keep containers in "monitoring" network
      - source_labels: [__meta_docker_network_name]
        regex: monitoring
        action: keep
      # Extract Port or overwrite `prometheus-port` label.
      - source_labels: [__address__]
        regex: '[^\s]+:(\d+)'
        target_label: __tmp_port
      - source_labels: [__meta_docker_container_label_prometheus_port]
        target_label: __tmp_port
        action: keepequal
```

 https://github.com/prometheus/prometheus/pull/10490#issuecomment-2283927339

**Comment by keslerm** [comment]
Will give this a go in the next day or so. Thanks for the help!

**Comment by f41gh7** [comment]
The feature was included into [v1.108.0](https://docs.victoriametrics.com/changelog/#v11080) release
